aicsimageio/readers/ome_tiff_reader.py

```diff
@@ -0,0 +1,584 @@
+            xml = ET.tostring(
+                root,
+                encoding="unicode",
+                method="xml",
+            )
+
+        return xml
+
+    @staticmethod
+    def _get_ome(ome_xml: str, clean_metadata: bool = True) -> OME:
+        # To clean or not to clean, that is the question
+        if clean_metadata:
+            ome_xml = OmeTiffReader._clean_ome_xml_for_known_issues(ome_xml)
+
+        return from_xml(ome_xml)
+
+    @staticmethod
+    def _is_supported_image(
+        fs: AbstractFileSystem, path: str, clean_metadata: bool = True
+    ) -> bool:
+        try:
+            with fs.open(path) as open_resource:
+                with TiffFile(open_resource) as tiff:
+                    # Get first page description (aka the description tag in general)
+                    xml = tiff.pages[0].description
+                    return OmeTiffReader._get_ome(xml, clean_metadata)
+
+        # tifffile exception, tifffile exception, ome-types / etree exception
+        except (TiffFileError, TypeError, ET.ParseError):
+            return False
+
+    def __init__(self, image: types.PathLike, clean_metadata: bool = True):
+        """
+        Wraps the tifffile and ome-types APIs to provide the same aicsimageio Reader
+        API but for volumetric OME-TIFF images.
+
+        Parameters
+        ----------
+        image: types.PathLike
+            Path to image file to construct Reader for.
+
+        clean_metadata: bool
+            Should the OME XML metadata found in the file be cleaned for known
+            AICSImageIO 3.x created errors.
+            Default: True (Clean the metadata for known errors)
+
+        Notes
+        -----
+        If the OME metadata in your file isn't OME schema compilant or does not validate
+        this will fail to read you file and raise an exception.
+
+        If the OME metadata in your file doesn't use the latest OME schema (2016-06),
+        this reader will make a request to the referenced remote OME schema to validate.
+        """
+        # Expand details of provided image
+        self.fs, self.path = io_utils.pathlike_to_fs(image, enforce_exists=True)
+        self.extension = ".".join(self.path.split(".")[1:])
+        self.clean_metadata = clean_metadata
+
+        # Enforce valid image
+        if not self._is_supported_image(self.fs, self.path, clean_metadata):
+            raise exceptions.UnsupportedFileFormatError(
+                self.__class__.__name__, self.extension
+            )
+
+    @property
+    def scenes(self) -> Tuple[str]:
+        if self._scenes is None:
+            with self.fs.open(self.path) as open_resource:
+                with TiffFile(open_resource) as tiff:
+                    ome = self._get_ome(tiff.pages[0].description, self.clean_metadata)
+                    self._scenes = tuple(image_meta.id for image_meta in ome.images)
+
+        return self._scenes
+
+    @staticmethod
+    def _process_ome_metadata(
```

Comment:
this could get a more specific name (like get_ome_dimensions?)

Reply:
Changed to `_get_dims_and_coords_from_ome`